ts/screens/profile/ProfileMainScreen.tsx

```diff
@@ -1,107 +1,97 @@
+import {
+  IOStackNavigationRouteProps,
+  useIONavigation
+} from "../../navigation/params/AppParamsList";
 import { MainTabParamsList } from "../../navigation/params/MainTabParamsList";
 import ROUTES from "../../navigation/routes";
 import { setDebugModeEnabled } from "../../store/actions/debug";
-import { navigateToLogout } from "../../store/actions/navigation";
-import { Dispatch } from "../../store/actions/types";
 import { isDebugModeEnabledSelector } from "../../store/reducers/debug";
-import { GlobalState } from "../../store/reducers/types";
 import { isDevEnv } from "../../utils/environment";
+import { useIODispatch, useIOSelector } from "../../store/hooks";
+import { showProfileBannerSelector } from "../../features/profileSettings/store/selectors";
+import { setShowProfileBanner } from "../../features/profileSettings/store/actions";
 import DeveloperModeSection from "./DeveloperModeSection";
 
+const consecutiveTapRequired = 4;
+const RESET_COUNTER_TIMEOUT = 2000 as Millisecond;
+
 type Props = IOStackNavigationRouteProps<MainTabParamsList, "PROFILE_MAIN"> &
   LightModalContextInterface &
-  ReturnType<typeof mapDispatchToProps> &
-  ReturnType<typeof mapStateToProps> &
   TabBarItemPressType;
 
-type State = {
-  tapsOnAppVersion: number;
-};
-
 type ProfileNavListItem = {
   value: string;
 } & Pick<
   ComponentProps<typeof ListItemNav>,
-  "description" | "testID" | "onPress" | "hideChevron"
+  "description" | "testID" | "onPress"
 >;
 
-const contextualHelpMarkdown: ContextualHelpPropsMarkdown = {
-  title: "profile.main.contextualHelpTitle",
-  body: "profile.main.contextualHelpContent"
-};
-
-const consecutiveTapRequired = 4;
-const RESET_COUNTER_TIMEOUT = 2000 as Millisecond;
+const ListItem = memo(ListItemNav);
 
 /**
  * A screen to show all the options related to the user profile
  */
-class ProfileMainScreen extends React.PureComponent<Props, State> {
-  private scrollViewContentRef = React.createRef<ScrollView>();
-
-  constructor(props: Props) {
-    super(props);
-    this.state = {
-      tapsOnAppVersion: 0
-    };
-  }
+const ProfileMainScreen = ({ setTabPressCallback, hideModal }: Props) => {
+  const dispatch = useIODispatch();
+  const navigation = useIONavigation();
+  const theme = useIOTheme();
+  const { show } = useIOToast();
+  const isDebugModeEnabled = useIOSelector(isDebugModeEnabledSelector);
+  const showProfileBanner = useIOSelector(showProfileBannerSelector);
+  const [tapsOnAppVersion, setTapsOnAppVersion] = useState(0);
+  const scrollViewContentRef = useRef<ScrollView>(null);
+  const idResetTap = useRef<number>();
 
-  public componentDidMount() {
-    this.props.setTabPressCallback(() => () => {
-      if (this.scrollViewContentRef.current) {
-        this.scrollViewContentRef.current.scrollTo({
+  useEffect(() => {
+    setTabPressCallback(() => () => {
+      if (scrollViewContentRef.current) {
+        scrollViewContentRef.current.scrollTo({
```

Comment:
I guess here we have a regression from [this](https://github.com/pagopa/io-app/pull/4422), but it's not due to this PR. Can we double check the reason why? If we are able to fix it in this PR, good. Otherwise we add a JIRA task we'll address this in a new PR. What do you think?

Reply:
fixed!